qemu/tests/x86_cpu_model.py

```diff
@@ -55,6 +46,56 @@ def run(test, params, env):
         guest_model = model_pattern % ""
     else:
         test.cancel("This host doesn't support cpu model %s" % model)
+    return cpu_model, guest_model, flags
+
+
+def check_flags(params, flags, test, session=None):
+    """
+    Check cpu flags
+    :param params: Dictionary with the test parameters
+    :param flags: checked flags
+    :param test: QEMU test object
+    :param session: guest session
+    """
+    if session:
+        if params["os_type"] == "linux":
+            error_context.context("Check cpu flags inside guest", logging.info)
+            cmd = params["check_flag_cmd"]
+            out = session.cmd_output(cmd).split()
+            missing = [f for f in flags.split() if f not in out]
+            if missing:
+                test.fail("Flag %s not in guest" % missing)
+            no_flags = params.get("no_flags")
+            if no_flags:
+                err_flags = [f for f in no_flags.split() if f in out]
+                if err_flags:
+                    test.fail("Flag %s should not be present in guest" % err_flags)
+    else:
+        check_host_flags = params.get_boolean("check_host_flags")
+        if check_host_flags:
+            cmd = params["check_host_flag_cmd"]
```

Comment:
What's the difference between the check cmd on host and guest?

Reply:
no difference, just for skipping windows because we don't define the check cmd for windows.